### DISCUSSION TITLE
Bei zu großen Images keine Fehlermeldung mehr und das Image wird erstellt (7530AX, 7590 und 5590 getestet)

### DISCUSSION THREAD

**Discussion by MasterRoCcO** [Freetz-NG/freetz-ng#1019] [Unerfüllte Wünsche] [answered]
Ich habe das schon vor Wochen bei ippf geschrieben. Aber angeblich hast du ja kein ACC mehr da!
Aber dann hier noch mal
Seit dem die Kabelboxen ja beim Image Bau immer gesagt haben, das sie zu groß sind, hat ja FDA das hier gemacht
https://github.com/Freetz-NG/freetz-ng/commit/6d46240c1a87b7f4e3d31d2ecd7aa98f0f8e7777
und
https://github.com/Freetz-NG/freetz-ng/commit/922c3fe1d23d1ac076707ed4248e1ece3a42d9dd
Aber das schöne ist nun, das auch andere Images mit Übergröße gebaut, werden man schaue auf max und free
```
STEP 3: PACK/SIGN
  checking for left over version-control-system files
  integrate freetz info file into image
packing var.tar
  checking signature key files
    adding public signature key file
creating filesystem image (SquashFS4_le-xz)
  SquashFS block size: 64 kB (65536 bytes)
copying kernel image
  kernel image size: 3.1 MB
copying filesystem image
  filesystem image size: 47.9 MB
packing fitimg file
  fitimg file size: 51.3 MB, max 50.0 MB, free -1.3 MB (-1326035 bytes)
packing images/7530AX_07.90.all-rev114104_inhaus_freetz-ng-24020M-7f717d502.image
  packed image file size: 51.4 MB (53872640 bytes)
signing packed .image file
  signed image file size: 51.4 MB (53872640 bytes)
source firmware: 7530-AX_de-es-fr-it-nl-pl 256.07.90 rev114104 {ALL} [Smart24 P1] (22.06.2024 01:23:26)
  source image file size: 42.4 MB (44441600 bytes)
done.

FINISHED
```
Keine Fehlermeldung und schön das Image erstellen. Bin gespannt, wie viele Leute nun versuchen werden übergroße Images in ihre Box zu Flashen

da wo free passt
```
STEP 3: PACK/SIGN
  checking for left over version-control-system files
  integrate freetz info file into image
packing var.tar
  checking signature key files
    adding public signature key file
creating filesystem image (SquashFS4_le-xz)
  SquashFS block size: 64 kB (65536 bytes)
copying kernel image
  kernel image size: 3.1 MB
copying filesystem image
  filesystem image size: 46.3 MB
packing fitimg file
  fitimg file size: 49.7 MB, max 50.0 MB, free 0.3 MB (357421 bytes)
packing images/7530AX_07.90.all-rev114104_inhaus_freetz-ng-24020M-7f717d502.image
  packed image file size: 49.8 MB (52193280 bytes)
signing packed .image file
  signed image file size: 49.8 MB (52193280 bytes)
source firmware: 7530-AX_de-es-fr-it-nl-pl 256.07.90 rev114104 {ALL} [Smart24 P1] (22.06.2024 01:23:26)
  source image file size: 42.4 MB (44441600 bytes)
done.

FINISHED
```

**Comment by fda77** [comment]
- Bei Puma6 wurde bislang die ARM größe für die Atom genommen, das wurde korrigiert.
- Für Puma7 gibt es den Wert nicht in avms config und es wurde ein statischer genommen der für alle aktuell verfügbaren Puma7 passen sollte
- Dein 3. Problem betrifft FIT Geräte: nach dem Packen gabe es noch NIE eine Fehlermeldung, nur die Info!
  Um ehrlich zu sein hab ich manche der Werte dafür auch nur geraten  (ohne Geräte .. du weisst)  dh es könnte auch gehen wenn negativ frei ist

Also alles in allem ein "feature request" und damit auf die auf die immerwährende Wish-List :)

**Reply by MasterRoCcO** [reply]
Gut das die 7590 aber mal kein FIT Gerät ist! 
Wenn das Image zu groß ist, wurde es früher nicht erstellt und wurde angebrochen!
Da man das zu große Image zwar hochladen. Aber bringt nichts. Da er nicht alles schreiben kann. Daher wird das alte Image gebootet.

**Reply by fda77** [reply]
Für die 7590 ich eigentlich nichts geändert. sicherß??
Am besten würde jeder auf seinem FIT schauen wieviel möglich ist, aber die freetz-community gibt das nicht her ^^

**Reply by MasterRoCcO** [reply]
Für die FIT auch nicht nur die Begrenzung Meldung hast du übergangen! Ansonsten könnte man ja nicht zu große Images erstellen.

**Reply by fda77** [reply]
nein, fit ist ein eigenes "verpackungsformat", da fehlt noch derjenige der diese meldung überhaupt erstellt (du?)

**Reply by MasterRoCcO** [reply]
Und warum darf ich dann bitte für die 7590 ein image mit 10 MB darüber bauen. Die schönen INHAUS Versionen dürfen dieses ja.

**Reply by fda77** [reply]
Hab ich Löcher in den Handflächen?

**Reply by MasterRoCcO** [reply]
Das kann ich dir nicht sagen! Ich sage dir nur das es seit den Updates nun mal geht!

**Reply by fda77** [reply]
Dir ist schon klar dass es für die 7590 keine inhaus in freetz gibt?
https://freetz-ng.github.io/freetz-ng/FIRMWARES.html#fritzbox-fon-wlan-75xx

**Reply by MasterRoCcO** [reply]
Nur, weil du es nicht in freetz-ng eingebaut hast heißt das nicht, dass es kein andere hat. Das sind nun mal nur 2 Zeilen und die /freetz-ng/fconfig/.img/generate.sh einmal nutzen

**Reply by fda77** [reply]
wegen "technischer Limitation" darf es nur ENTWERDER inhaus ODER labor .image-file in der .in Datei geben.
Vielleicht ists deshalb

**Reply by MasterRoCcO** [reply]
Ich kann dir nur sagen, dass es genau seit dem Aufheben das Limit bei den Cabel Boxen so ist. Ist mir auch mehr als Zufall aufgefallen. Da normal die meisten neuen Boxen hat genug Space haben. Aber es gibt halt so ein paar wie zB die 7530AX die mehr und mehr an ihr Ende des Spaces kommt. Genauso wie die 6850xx Geräte, viel wird da nicht mehr gehen seitens AVM. 

Und nun mal was zur `config/.img/generate.sh` da solltest du mal das Einbauen
```
@@ -475,6 +475,20 @@
 	[ "$X" == "available" ] && in_b "FREETZ_AVM_HAS_${P^^}"
 	[ $DOSHOW -ge 2 ] && outp "${P,,}" "$X"
 
+	#ZIGBEE
+	P=ZIGBEE
+	X="$(sed -rn "s/^export CONFIG_$P=\"?([^\"])\"?.*$/\1/p" "$unpacked/etc/init.d/rc.conf" | head -n1)"
+	[ "$X" != "y" ] && X="%" || X="available"
+	[ "$X" == "available" ] && in_b "FREETZ_AVM_HAS_${P^^}"
+	[ $DOSHOW -ge 2 ] && outp "${P,,}" "$X"
+	
+	#MATTER
+	P=MATTER
+	X="$(sed -rn "s/^export CONFIG_$P=\"?([^\"])\"?.*$/\1/p" "$unpacked/etc/init.d/rc.conf" | head -n1)"
+	[ "$X" != "y" ] && X="%" || X="available"
+	[ "$X" == "available" ] && in_b "FREETZ_AVM_HAS_${P^^}"
+	[ $DOSHOW -ge 2 ] && outp "${P,,}" "$X"
+
 	#FHEM
 	P=FHEM
 	X="$(sed -rn "s/^export CONFIG_$P=\"?([^\"])\"?.*$/\1/p" "$unpacked/etc/init.d/rc.conf" | head -n1)"
```
Da Matter wird zwar erst in der Labot von dem Gate genutzt wird aber früher oder später bei anderen auch kommen. ZIGBEE kann ja die 5690Pro, 6670 7690 und der GATE

**Reply by fda77** [reply]
Vielleicht ist irgend wo ein Fehler... ich kann aber heut kein gosses image mehr bauen, bzw über haupt eins. Und mit der Labor gibts das Problem nicht?
MATTER und ZIGBEE brauch man nicht, solang nicht irgend eine andere Funktion die AVM_HAS variable braucht, zb ein remove patch ist nur sichtbar wenn die firmware die komponenten hat

**Reply by MasterRoCcO** [reply]
Ich kann doch nicht alles immer testen. Los ein Image bauen geht doch einfach. 

Lieber jetzt schon als später ;-). Hast ja auch andere Sachen nicht eingebaut, aber sind in den *.in Dateien. Aber soll mir auch egal sein. Da ich freetz-ng nur auf der Spielbox nutze (7530AX) alles andere ist schon seit Monaten freetz-ng FREE.

Bin nun auch raus. Morgen ist ein langer Tag im Büro.

**Reply by fda77** [reply]
Ich hab aber hier keine flotte CPU dafür - und ich denke nicht dass es ein fehler ist

**Reply by MasterRoCcO** [reply]
https://www.youtube.com/watch?v=sCJu71JxmbE

Mein Prof. sagte mir damals immer. Denken ist nicht wissen, liefer den gegen Beweis. 
Dann dauert der Bau halt 1 H wo ist das Prob. Kann man doch schön nebenbei laufen lassen.

**Reply by fda77** [reply]
Das war dann aber nicht der für Stochastik. 
Der hätte es auch für "unwahrscheinlich" gehalten dass du recht hast

**Reply by fda77** [reply]
So, dann poste mal die .config mit der der fehler auftritt

https://github.com/Freetz-NG/freetz-ng/blob/master/config/avm/features.in#L161

**Reply by fda77** [reply]
Jetzt hab ich das mit matter+zigbee verstanden! Die sind aus versehen reingerutscht: 8c21736f9056ec85ba49fc900a2c5120a814b82f + https://github.com/Freetz-NG/freetz-ng/actions/runs/10165491590/job/28113397630

**Reply by MasterRoCcO** [reply]
NETT 6670 7690 5690PRO GATE  das sie kein zigbee können

**Reply by MasterRoCcO** [reply]
Mein Prof würde dich lieben. Da er dir zeigen würde das du nicht das glauben kannst was du denkst.

**Reply by fda77** [reply]
Mit Genderstudies Leuten komme ich nicht soo gut klar...
Ich warte übrigens noch immer auf deine 7590 config!

**Reply by fda77** [reply]
Also hat die 7590 dann doch kein Problem mehr?

Machst du die Fehler- (oder Warn-)Meldung für fitimg?

**Reply by MasterRoCcO** [reply]
Und warum willst du eine config. Dir sollte klar sein, dass es über die config nicht so einfach sein sollte ein zu großes Image zu erstellen. Es gibt da auch noch andere Tricks, ein zu großes Image zu erstellen.

**Reply by fda77** [reply]
Okay, wie du meinst. Bei der 7590 gibt es also kein Problem.
Es bleibt damit beim "featrure-request" für "fitimg"

**Comment by fda77** [comment, marked as answer]
-> 2f25b78eb
